common/djangoapps/student/tests/tests.py

```diff
@@ -485,19 +486,33 @@ def test_linked_in_add_to_profile_btn_with_certificate(self):
         self.assertContains(response, expected_url)
 
 
-class EnrollmentEventTestMixin(object):
-    """ Mixin with assertions for validating enrollment events. """
-
+class UserSettingsEventTestMixin(EventTestMixin):
+    """
+    Mixin for verifying that user setting events were emitted during a test.
+    """
     def setUp(self):
-        super(EnrollmentEventTestMixin, self).setUp()
-        patcher = patch('student.models.tracker')
-        self.mock_tracker = patcher.start()
-        self.addCleanup(patcher.stop)
+        super(UserSettingsEventTestMixin, self).setUp('util.model_utils.tracker')
 
-    def assert_no_events_were_emitted(self):
-        """Ensures no events were emitted since the last event related assertion"""
-        self.assertFalse(self.mock_tracker.emit.called)  # pylint: disable=maybe-no-member
-        self.mock_tracker.reset_mock()
+    def assert_user_setting_event_emitted(self, **kwargs):
+        """
+        Helper method to assert that we emit the expected user settings events.
+
+        Expected settings are passed in via `kwargs`.
+        """
+        if 'truncated' not in kwargs:
+            kwargs['truncated'] = []
+        self.assert_event_emitted(
```

Comment:
No-- tests have been handling that themselves (generally in setup, and in specific places within a test where desired). It might be nice to have this method do it, but I don't think we have time for that refactoring.